tesseract-server/src/handlers/metadata.rs

```diff
@@ -198,7 +207,38 @@ pub fn do_members(
 }
 
 
+pub fn apply_filter_term(df: DataFrame, term: &str) -> DataFrame {
+    debug!("Filtering dataframe by: {}", term);
+
+    // Collect all column names with text data
+    let columns: Vec<String> = (&df.columns).iter()
+        .filter_map(|column| match column.column_data {
+            ColumnData::Text(_) => Some(column.name.clone()),
+            ColumnData::NullableText(_) => Some(column.name.clone()),
+            _ => None,
+        })
+        .collect();
+
+    df.drain_filter(|datum, _| {
```

Comment:
It's a bit confusing to me that this param name is `datum` when it's a HashMap. And can the second param be removed?